conbench/entities/commit.py

```diff
@@ -532,6 +534,62 @@ def __init__(self):
             "testing repository with git@g",
         ]
 
+    def _read_tokens_from_env(self) -> None:
+        """
+        This reads GITHUB_API_TOKEN, initializes a cycling iterator (if more
+        than one token was provided), and always populates self._token_pool
+        (None, or pool of size greater 1), and self._current_auth_token (with
+        either None, or with a token).
+
+        When reading GitHub API token(s) from environment: support two formats:
+        one token, or more than one token (comma-separated)
+        """
+
+        data = os.getenv("GITHUB_API_TOKEN")
+        self._token_pool: Optional[itertools.cycle[str]] = None
+        self._current_auth_token = None
+        if data is None:
+            log.info("GITHUB_API_TOKEN env not set")
+            return
+
+        log.info("GITHUB_API_TOKEN env was set, length of data: %s", len(data))
+
+        token_candidates = [data]
+        if "," in data:
+            token_candidates = data.split(",")
+
+        tokens_to_use = []
+        for t in token_candidates:
+            if len(t) < 5 or len(t) > 120:
+                log.info("bad token length, ignore: %s", len(t))
+            else:
+                tokens_to_use.append(t)
+
+        if len(tokens_to_use) == 0:
+            return
+
+        if len(tokens_to_use) == 1:
+            self._current_auth_token = tokens_to_use[0]
+            log.info("configured one GitHub HTTP API auth token")
+            return
+
+        self._token_pool = itertools.cycle(tokens_to_use)
+        self._current_auth_token = next(self._token_pool)
+        log.info("current auth token starts with: %s", self._current_auth_token[:3])
+
+    def _rotate_auth_token(self):
+        """
+        Return True if token was rotated, False otherwise.
+        """
+        if self._token_pool is None:
+            # convention: pool size greater than 1, rotation takes effect.
+            log.debug("tried rotating auth token, but no pool is configured")
+            return False
+
+        self._current_auth_token = next(self._token_pool)
+        log.info("current auth token starts with: %s", self._current_auth_token[:3])
```

Comment:
nit: I think personal access tokens always start with `ghp_`. Maybe log the last three chars instead?

Reply:
yes! opted for still showing prefix, just a tiny bit longer